lib/printer/printer.cpp

```diff
@@ -4,142 +4,244 @@
 
 #include <map>
 
+namespace {
+
+using namespace kllvm;
+using namespace kllvm::parser;
+
 const std::string WHITESPACE = " \n\r\t\f\v";
 
-static std::string ltrim(const std::string &s) {
+std::string ltrim(const std::string &s) {
   size_t start = s.find_first_not_of(WHITESPACE);
   return (start == std::string::npos) ? "" : s.substr(start);
 }
 
-static std::string rtrim(const std::string &s) {
+std::string rtrim(const std::string &s) {
   size_t end = s.find_last_not_of(WHITESPACE);
   return (end == std::string::npos) ? "" : s.substr(0, end + 1);
 }
 
-static std::string trim(const std::string &s) {
+std::string trim(const std::string &s) {
   return rtrim(ltrim(s));
 }
 
-namespace kllvm {
+std::map<std::string, std::string> getFormats() {
+  static std::map<std::string, std::string> formats;
+  static bool once = false;
```

Comment:
same comment here about once; we can flip the values and it becomes a bit easier to read. Not going to comment every time this appears in the PR after this though.